pkg/block/block.go

```diff
@@ -103,7 +103,7 @@ func Upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir st
 		return errors.Wrap(err, "encode meta file")
 	}
 
-	if err := bkt.Upload(ctx, path.Join(DebugMetas, fmt.Sprintf("%s.json", id)), bytes.NewReader(metaEncoded.Bytes())); err != nil {
+	if err := bkt.Upload(ctx, path.Join(DebugMetas, fmt.Sprintf("%s.json", id)), &metaEncoded); err != nil {
```

Comment:
We are reading from `&metaEncoded` before this line hence the offset will be at the end. Or IOW this line will not get the data. The "correct" fix here is probably to use `strings.Builder` and then do `strings.NewReader` (:

Reply:
Thanks for the investigation! Let me fix this.

Reply:
Hi @GiedriusS, thanks for the help and the CI passed. Please take a look again when you have time.